src/pep8ext_naming.py

```diff
@@ -65,10 +65,12 @@ def __init__(cls, class_name, bases, namespace):
 def _err(self, node, code, **kwargs):
     lineno, col_offset = node.lineno, node.col_offset
     if isinstance(node, ast.ClassDef):
-        lineno += len(node.decorator_list)
+        if PYTHON_VERSION < (3, 8):
+            lineno += len(node.decorator_list)
```

Comment:
a note that even in python<3.8 this is potentially incorrect as decorators can be an arbitrary number of lines:

```python
@foo(
    bar,
)
@baz
def F(): pass
```

Reply:
Yes, the previous approach was certainly a bit of a kludge. I'm glad this is fixed at the AST level now.